frontends/web/src/app.tsx

```diff
@@ -116,14 +113,33 @@ class App extends Component<Props, State> {
             }
         });
 
-        getAccounts()
-            .then(accounts => this.setState({ accounts }, this.maybeRoute))
-            .catch(console.error);
+        getDeviceList()
+        .then(devices => {
+            return getAccounts()
+            .then(accounts => {
+                this.setState({ accounts, devices }, this.maybeRoute);
+            });
+        })
+        .catch(console.error);
 
         this.unsubscribeList.push(
             syncAccountsList(accounts => {
                 this.setState({ accounts }, this.maybeRoute);
             }),
+            syncDeviceList(devices => {
+                const oldDeviceIDList = Object.keys(this.state.devices);
```

Comment:
Scattering routing logic makes it harder to reason about the routing conditions and rules.

Why not simply call `this.maybeRoute` like above with the accounts, and keep the routing code there?

Reply:
without this PR the app routes many times back and forth and after routing happen routes again and was really hard to follow.

This PR reduces the amount of routing and depends on checking if there is a new device, this can't be done in  maybeRoute.

Also there are other arguments against maybeRoute see my other comment.